File: Exercism/csharp/log-levels/LogLevels.cs
using System;
using System.Text.RegularExpressions;

static class LogLine
{   
    public static string Message(string logLine) => new Regex(@"\s+(?<message>[\w\s]+)").Match(logLine).Groups["message"].Value.Trim();
    public static string LogLevel(string logLine) => new Regex(@"\[(?<level>\w+)\]").Match(logLine).Groups["level"].Value.ToLowerInvariant();
    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: Exercism/fsharp/log-levels/LogLevels.fs
module LogLevels

open System.Text.RegularExpressions

let message logLine = Regex.Match(logLine, "\]:\s+(?<message>.+)\s*").Groups.["message"].Value.Trim()
let logLevel logLine = Regex.Match(logLine, "\[(?<level>\w+)\]").Groups.["level"].Value.ToLowerInvariant()
let reformat logLine = $"{message logLine} ({logLevel logLine})"
